### ISSUE TITLE
Add a --dryrun flag to the `az storage remove` command

### ISSUE THREAD

**Issue by alexwlchan** [Azure/azure-cli#14875]
**Is your feature request related to a problem? Please describe.**
I'm a new user trying to delete some blobs in a container. I've never used the Azure CLI before, and I'm a bit nervous.

Being able to see what's going to happen before I run a destructive operation like `az storage remove` would be helpful, and give me confidence I'm not about to break something.

**Describe the solution you'd like**
A new flag `--dryrun` that prints the list of blobs to be deleted, but doesn't actually

**Describe alternatives you've considered**
Writing my own code to do the deletion, but that seems way more work for something I can probably do with the CLI.

**Additional context**
There's a similar flag in `az storage blob delete-batch` (although this disclaimer isn't reassuring: _So the file(s) will be deleted with the command without --dryrun may be different from the result list with --dryrun flag on._).

**Comment by alexwlchan** [comment]
Spoiler alert: yeah, I got this wrong, and accidentally deleted everything in the container.

It's not a disaster because I was in a staging environment and everything can be recreated, but having a `--dryrun` command would have helped.

**Comment by Juliehzl** [comment]
@zezha-msft could you help prioritize the feature request for azcopy?

**Comment by zezha-msft** [comment]
Hi @Juliehzl, thanks for letting us know! I've entered a PBI into our backlog to be prioritized.

Hi @alexwlchan, to clarify, the service is the only source of truth, so even if we implemented this dry run option, its results may still be different from the actual delete operation, since there may be changes in between when the dry run is executed and the actual delete happens, e.g. a new blob got added to the container. Is this within your expectations?

**Comment by alexwlchan** [comment]
> the service is the only source of truth, so even if we implemented this dry run option, its results may still be different from the actual delete operation, since there may be changes in between when the dry run is executed and the actual delete happens, e.g. a new blob got added to the container. Is this within your expectations?

Yes, having thought about it some more, this makes more sense to me than when I originally opened the issue.

I'd mostly be using it to check the broad shape of the command – e.g. _"I expect to delete everything starting with `a/b/c`_", and checking the dry run output doesn't include unrelated objects.

**Comment by zezha-msft** [comment]
Sounds good @alexwlchan, and just to confirm, you expect the output of the command to be the whole list of blobs that will be deleted? That can get a bit crazy if there are thousands of entries. I guess you could pump it into a file and then perform a search.

**Comment by alexwlchan** [comment]
> you expect the output of the command to be the whole list of blobs that will be deleted? 

Without giving it too much thought, yes. I'd usually dump it to a file or `less`.

**Comment by zezha-msft** [comment]
Thanks for the info @alexwlchan. We'll discuss the general UX a bit more on our side and schedule the feature accordingly.
